core/src/main/java/nl/nn/adapterframework/jndi/JndiDataSourceFactory.java

```diff
@@ -15,38 +15,85 @@
 */
 package nl.nn.adapterframework.jndi;
 
+import java.sql.Connection;
+import java.time.Duration;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Properties;
 
 import javax.naming.NamingException;
 import javax.sql.CommonDataSource;
 import javax.sql.DataSource;
-
+import javax.sql.XADataSource;
+
+import org.apache.commons.dbcp2.ConnectionFactory;
+import org.apache.commons.dbcp2.DataSourceConnectionFactory;
+import org.apache.commons.dbcp2.PoolableConnection;
+import org.apache.commons.dbcp2.PoolableConnectionFactory;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.pool2.ObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPool;
 import org.springframework.jdbc.datasource.TransactionAwareDataSourceProxy;
 
+import lombok.Getter;
+import lombok.Setter;
 import nl.nn.adapterframework.jdbc.IDataSourceFactory;
 import nl.nn.adapterframework.util.AppConstants;
 
 /**
  * Factory through which (TX-enabled) DataSources can be retrieved.
- * 
+ *
  * Already created DataSources are stored in a ConcurrentHashMap.
  * Every DataSource can be augmented before it is added.
  */
 public class JndiDataSourceFactory extends JndiObjectFactory<DataSource,CommonDataSource> implements IDataSourceFactory {
 
 	public static final String DEFAULT_DATASOURCE_NAME_PROPERTY = "jdbc.datasource.default";
 	public static final String GLOBAL_DEFAULT_DATASOURCE_NAME = AppConstants.getInstance().getProperty(DEFAULT_DATASOURCE_NAME_PROPERTY);
+	@Getter @Setter protected int minPoolSize = 0;
+	@Getter @Setter protected int maxPoolSize = 20;
+	@Getter @Setter protected int maxIdle = 2;
+	@Getter @Setter protected int maxLifeTime = 0;
+	@Getter @Setter protected int connectionCheckInterval = 300;
+	@Getter @Setter protected String testQuery = null;
 
 	public JndiDataSourceFactory() {
 		super(CommonDataSource.class);
+		AppConstants appConstants = AppConstants.getInstance();
+		minPoolSize = appConstants.getInt("transactionmanager.jdbc.connection.minPoolSize", minPoolSize);
+		maxPoolSize = appConstants.getInt("transactionmanager.jdbc.connection.maxPoolSize", maxPoolSize);
+		maxIdle = appConstants.getInt("transactionmanager.jdbc.connection.maxIdle", maxIdle);
+		maxLifeTime = appConstants.getInt("transactionmanager.jdbc.connection.maxLifeTime", maxLifeTime);
+		connectionCheckInterval = appConstants.getInt("transactionmanager.jdbc.connection.checkInterval", connectionCheckInterval);
+		testQuery = appConstants.getString("transactionmanager.jdbc.connection.testQuery", testQuery);
 	}
 
 	protected DataSource augmentDatasource(CommonDataSource dataSource, String dataSourceName) {
+		if (dataSource instanceof XADataSource) {
+			return createXAPool((XADataSource) dataSource, dataSourceName);
+		}
```

Comment:
Deze moet ook in de maxPoolSize if check?

Reply:
Ik hernoem de method naar `createXADataSource()` om verwarring te voorkomen. Bij BTM wordt al altijd een pool gecreerd voor XA sessions, ongeacht maxPoolSize. Die zet daarin ook een aantal andere properties, die niet direct gerelateerd zijn aan pooling, dus ik durf daar niet teveel aan te veranderen.

Met een andere naam van de method is het hopelijk minder verwarrend.